server/src/index.ts

```diff
@@ -3,11 +3,125 @@
 // SPDX-License-Identifier: MIT
 
 import express, { Application, Request, Response } from 'express';
+import { validateLogin, performSignup, PlainPassword, UserRole } from './auth';
+import cors from 'cors';
 
 const app: Application = express();
 const parsedPort = Number(process.env.AALTO_GRADES_BACKEND_PORT);
 const port: number = isNaN(parsedPort) ? 3000 : parsedPort;
 
+interface LoginRequest {
+  username: string,
+  password: PlainPassword,
+}
+
+interface SignupRequest {
+    username: string,
+    password: PlainPassword,
+    email: string,
+    role: UserRole,
+}
+
+const validateUserRole = (role: any): role is UserRole => typeof role === 'string' && (
+  role === 'Teacher' ||
+    role === 'Student' ||
+    role === 'Admin'
+);
+
+const validateLoginFormat = (body: any): body is LoginRequest =>
+  body &&
+    body.username &&
+    body.password &&
+    typeof body.username === 'string' &&
+    typeof body.password === 'string';
+
+const validateSignupFormat = (body: any): body is SignupRequest =>
+  body &&
+    body.username &&
+    body.password &&
+    body.email &&
+    validateUserRole(body.role) &&
+    typeof body.username === 'string' &&
+    typeof body.password === 'string' &&
+    typeof body.email === 'string';
```

Comment:
We have decided to use the function keyword for named functions, and arrow functions for anonymous functions. These functions should be changed so that they use the function keyword.

Reply:
Fixed up!